lib/jsonapi/authorization/authorizing_processor.rb

```diff
@@ -181,27 +186,32 @@ def authorize_replace_to_many_relationship
         )
       end
 
-      def authorize_remove_to_many_relationship
+      def authorize_remove_to_many_relationships
         source_resource = @resource_klass.find_by_key(
           params[:resource_id],
           context: context
         )
         source_record = source_resource._model
 
-        relationship_type = params[:relationship_type].to_sym
-        related_resource = @resource_klass
-          ._relationship(relationship_type)
+        related_resources = @resource_klass
+          ._relationship(params[:relationship_type].to_sym)
           .resource_klass
-          .find_by_key(
-            params[:associated_key],
+          .find_by_keys(
+            params[:associated_keys],
             context: context
           )
-        related_record = related_resource._model unless related_resource.nil?
+
+        # find_by_keys doesn't raise if no records, so raise explicitly here
+        if related_resources.empty?
+          fail JSONAPI::Exceptions::RecordNotFound, params[:associated_keys]
```

Comment:
Hmm I guess we'd really want to test that the `DELETE` operation authorizes properly, and doesn't use 404 when the comments can't be found.

Can you modify the tests to expect that `remove_to_many_relationship` authorizer method will be called with the proper arguments?